src/test/Settings/CredentialsStorage.test.tsx

```diff
@@ -0,0 +1,110 @@
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+import * as React from 'react';
+import renderer, { act } from 'react-test-renderer';
+import { render, cleanup, screen, waitFor, getByText } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { CredentialsStorage } from '@app/Settings/CredentialsStorage';
+
+import { getFromLocalStorage, saveToLocalStorage } from '@app/utils/LocalStorage';
+
+jest.mock('@app/utils/LocalStorage', () => {
+  const map = new Map<any, any>();
+  return {
+    getFromLocalStorage: jest.fn((key: any, defaultValue: any): any => {
+      if (map.has(key)) {
+        return map.get(key);
+      }
+      return defaultValue;
+    }),
+
+    saveToLocalStorage: jest.fn((key: any, value: any): any => {
+      map.set(key, value);
+    }),
+  };
+});
+
+const storageKey = 'JMX_CREDENTIAL_LOCATION';
+const sessionStorageValue = 'Session (Browser Memory)';
+const backendStorageValue = 'Backend';
+
+describe('<CredentialsStorage/>', () => {
+  afterEach(cleanup);
+
+  it('renders correctly', async () => {
+    let tree;
+    await act(async () => {
+      tree = renderer.create(React.createElement(CredentialsStorage.content, null));
+    });
+    expect(tree.toJSON()).toMatchSnapshot();
+  });
+
+  it('defaults to Session storage', async () => {
+    render(React.createElement(CredentialsStorage.content, null));
+
+    expect(getFromLocalStorage).toHaveBeenCalledTimes(1);
+    expect(saveToLocalStorage).toHaveBeenCalledTimes(1);
+    expect(saveToLocalStorage).lastCalledWith(storageKey, sessionStorageValue);
+
+    expect(screen.getByText(sessionStorageValue)).toBeVisible();
+    expect(screen.queryByText(backendStorageValue)).toBeFalsy();
+  });
+
+  it('sets value to local storage when dropdown is clicked', async () => {
+    render(React.createElement(CredentialsStorage.content, null));
+
+    expect(getFromLocalStorage).toHaveBeenCalledTimes(1);
+    expect(saveToLocalStorage).toHaveBeenCalledTimes(1);
+    expect(saveToLocalStorage).lastCalledWith(storageKey, sessionStorageValue);
+
+    userEvent.click(screen.getByRole('button'));
+
+    const ul = await screen.findByRole('listbox');
+    const session = getByText(ul, sessionStorageValue);
+    const backend = getByText(ul, backendStorageValue);
+    userEvent.click(backend);
+    await waitFor(() => expect(session).not.toBeVisible());
```

Comment:
I am thinking this should wait for the menu to close as it is closer to what the user would see and expect?